File: python-impl/insfrastructure/usecases/create_account_usecase.py
from insfrastructure.entities.person_entity import PersonEntity
from insfrastructure.enums.account_types import AccountType
from insfrastructure.repositories.bank_account_repository import BankAccountRepository


class CreateAccountUseCase:
    def __init__(self, bank_account_repo: BankAccountRepository):
        self.bank_account_repo = bank_account_repo

    def execute(self, account_holder: PersonEntity, initial_deposit: float, account_type: AccountType = AccountType.SAVINGS):
        return self.bank_account_repo.create(account_holder, initial_deposit, account_type)

File: typescript-impl/src/infrastructure/usecases/create_account_usecase.ts
import { BankAccountEntity } from '../entities/bank_account_entity';
import { PersonEntity } from '../entities/person_entity';
import { AccountType } from '../enums/account_type';
import { BankAccountRepository } from '../repositories/bank_account_repository';

export class CreateAccountUseCase {
    constructor(private bank_account_repo: BankAccountRepository) { }

    execute(account_holder: PersonEntity, initial_deposit: number, account_type: AccountType = AccountType.SAVINGS): BankAccountEntity {
        return this.bank_account_repo.create(account_holder, initial_deposit, account_type);
    }
}
